src/internal/memory/transient_pool.hpp

```diff
@@ -0,0 +1,148 @@
+/**
+ * SPDX-FileCopyrightText: Copyright (c) 2022, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include "srf/data/reusable_pool.hpp"
+#include "srf/memory/buffer.hpp"
+#include "srf/memory/resources/memory_resource.hpp"
+#include "srf/utils/macros.hpp"
+
+#include <glog/logging.h>
+
+#include <cstddef>
+#include <new>
+#include <utility>
+
+namespace srf::internal::memory {
+
+/**
+ * @brief A short-lived buffer based on a portion of a data::SharedResable<memory::buffer>
+ *
+ * @see TransientPool for more details.
+ */
+
+class TransientBuffer
+{
+  public:
+    TransientBuffer(void* addr, std::size_t bytes, srf::data::SharedReusable<srf::memory::buffer> buffer);
+    ~TransientBuffer();
+
+    DELETE_COPYABILITY(TransientBuffer);
+    DEFAULT_MOVEABILITY(TransientBuffer);
+
+    void* data();
+    std::size_t bytes() const;
```

Comment:
Maybe just a comment of what bytes represents here, is it the size of `data` or is it the total size of `TransientBuffer`>

Also is `addr` null terminated?